src/mimecodec.ts

```diff
@@ -323,13 +350,18 @@ export function headerLinesDecode (headers) {
     const key = header.key.toLowerCase()
     const value = header.value
 
-    if (!headersObj[key]) {
-      headersObj[key] = value
+    if (!headersObjArr[key]) {
+      headersObjArr[key] = [value]
     } else {
-      headersObj[key] = [].concat(headersObj[key], value)
+      headersObjArr[key].push(value)
     }
   }
 
+  // convert single value arrays to single values
+  const headersObj = Object.fromEntries(
```

Comment:
It was hard to work around TS requirements, and it was simpler to do another pass